### PR TITLE
review validations for creating defaults and also fix saving periods

### PR DESCRIPTION
**Pull request by Baalmart**
# review validations for creating defaults and also fix saving periods

**_WHAT DOES THIS PR DO?_**
review validations for creating defaults and also fix saving periods

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-992]

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/hotfix-chart-defaults-PLAT-992

**_HOW DO I TEST OUT THIS PR?_**
README file
The ENVs and config files are here: https://drive.google.com/drive/folders/1rW4p1l4aD4zLuigQXOhJN7d8tF3M7n9J
Config files are under config sub directory

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [Create defaults](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/auth-service/create-defaults#create-defaults)
- [x] [Update defaults](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/auth-service/create-defaults#update-default)

[PLAT-992]: https://airqoteam.atlassian.net/browse/PLAT-992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/auth-service/routes/api-v1.js [hunk 1]
```diff
@@ -331,48 +349,54 @@ router.post(
...
       body("frequency")
-        .if(body("frequency").exists())
-        .notEmpty()
+        .exists()
+        .withMessage("frequency is is missing in your request")
+        .bail()
         .trim()
         .toLowerCase()
-        .isIn(["daily", "hourly", "monthly"])
+        .isIn(["daily", "hourly", "monthly, diurnal"])
```
Comment (Baalmart) [inline src/auth-service/routes/api-v1.js:366]: the `diurnal `validation has been added from here.... @ja-odur

File: src/auth-service/models/Defaults.js [hunk 2]
```diff
@@ -5,6 +5,16 @@ const { logElement, logText, logObject } = require("../utils/log");
 const isEmpty = require("is-empty");
 const HTTPStatus = require("http-status");
 
+const periodSchema = new mongoose.Schema(
+  {
```
Comment (Baalmart) [inline src/auth-service/models/Defaults.js:9]: Schema has been update to reflect the sub-fields for the period field. @ja-odur

File: src/auth-service/routes/api-v1.js [hunk 3]
```diff
@@ -290,7 +290,25 @@ router.put(
           return ObjectId(value);
         }),
       body("chartTitle").if(body("chartTitle").exists()).notEmpty().trim(),
-      body("period").if(body("period").exists()).notEmpty().trim(),
+      body("period")
+        .if(body("period").exists())
+        .notEmpty()
```
Comment (Baalmart) [inline src/auth-service/routes/api-v1.js:295]: For the update of defaults endpoint (PUT), this particular validation for the` period `field is misbehaving. But shall continue investigating it tomorrow. @ja-odur

**Review by Baalmart — COMMENTED** [review]
just adding some guiding notes for your review exercise. @ja-odur

File: src/auth-service/routes/api-v1.js (hunk 1 above)
Comment (Baalmart) [reply src/auth-service/routes/api-v1.js:364]: Resolved

File: src/auth-service/routes/api-v1.js (hunk 3 above)
Comment (Baalmart) [reply src/auth-service/routes/api-v1.js:295]: this was also resolved

**Review by 123MwanjeMike — COMMENTED** [review]
Great stuff. Just a minor concern tho

File: src/auth-service/routes/api-v1.js [hunk 4]
```diff
@@ -381,39 +403,62 @@ router.post(
...
+        .withMessage("period.unitValue is is missing in your request")
+        .bail()
+        .isFloat()
+        .withMessage("period.unitValue must be a number"),
+      body("period.label")
+        .exists()
+        .withMessage("period.label is is missing in your request"),
+      body("period.value")
+        .exists()
+        .withMessage("period.value is is missing in your request"),
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/auth-service/routes/api-v1.js:444]: Yeah, this is a typo. Let me fix it. Thanks!

[the remark this answers is not in the document]
Comment (Baalmart) [reply src/auth-service/routes/api-v1.js]: This has been addressed in my most recent commit. Thanks again!

**Review by 123MwanjeMike — APPROVED** [review]
LGTM

**Review by ja-odur — CHANGES_REQUESTED** [review]
@Baalmart The update endpoint is not recognising `diurnal` as a valid frequency
image

**Comment by Baalmart** [comment]
> @Baalmart The update endpoint is not recognising `diurnal` as a valid frequency image

This has been addressed, please PULL and proceed. @ja-odur

**Review by ja-odur — APPROVED** [review]
LGTM

**Pull request merged**